lib/src/functions/math.dart

```diff
@@ -98,3 +101,57 @@ BuiltInCallable _numberFunction(String name, num transform(num value)) {
         denominatorUnits: number.denominatorUnits);
   });
 }
+
+final _clamp = BuiltInCallable("clamp", r"$min, $preferred, $max", (arguments) {
+  var min = arguments[0].assertNumber("min");
+  var preferred = arguments[1].assertNumber("preferred");
+  var max = arguments[2].assertNumber("max");
+
+  if (min.hasUnits != preferred.hasUnits ||
+      preferred.hasUnits != max.hasUnits) {
+    throw SassScriptException(
+        "Mixing unitless arguments with arguments that have units is not allowed.");
```

Comment:
It would be good to provide more detail here. Which numbers are in conflict? What are their values? The user may be confused about how they got into this state, and this information could help them figure it out.

Reply:
Hmm good question. In this context it could make sense to have the majority "win out"—so if `arg1` and `arg2` are unitless but `arg3` has units, the error could be `Error: Expected $arg3 to have no units`.

This error might be confusing for a call to `hypot()` with lots of args (e.g. 4 unitless and 5 unitful) but that seems like the uncommon case.

Reply:
Actually, thinking about it more, if the user reads `Error: Expected $arg3 to have no units`, in the future they might assume that `clamp` only takes unitless values.

Reply:
How about—`Error: $arg1 has units but $arg2 does not. Mixing unitless arguments with arguments that have units is not allowed.` It's wordier than most messages we currently have, but it feels pretty clear.